src/geo/transform.js

```diff
@@ -185,6 +190,52 @@ class Transform {
         this._calcMatrices();
     }
 
+    get padding(): EdgeInsetJSON { return this._edgeInsets.toJSON(); }
```

Comment:
Serving the existing API in native, i didn't like that in Android we use "padding" and in iOS "edgeInsets". is padding here added to keep it close to native implementation?

Reply:
I just did it so the public facing term for is it simpler, but happy to change it to whatever.
Maybe @chloekraw , do you have any thoughts?

Reply:
For this feature, I think `padding` is the term that's going to be closest to what web developers who use CSS would expect. `inset` is most commonly used as a type of border style:
- https://developer.mozilla.org/en-US/docs/Web/CSS/padding
- https://developer.mozilla.org/en-US/docs/Web/CSS/border-style

I did find this https://developer.mozilla.org/en-US/docs/Web/CSS/inset "experimental technology" but it sounds like this `inset` is more about defining margins around blocks of text and is less analogous to a "viewport." (related: https://developer.mozilla.org/en-US/docs/Web/CSS/inset-block, https://developer.mozilla.org/en-US/docs/Web/CSS/inset-inline)

@arindam1993 are there other parts of the GL-JS code base that also use this idea of padding?

Reply:
@arindam1993 , related to this - what's your opinion about propagating CSS style padding specified to Map's parent HTML container to transform?

Reply:
@chloekraw I don't think so.

@astojilj I don't think we should, I think devs waning to create floating UI elements, which this feature is designed for would want to set a `padding` on the container, since it'd offset all their UI elements which I assume would be appended as children fo the `map` container.

However I think it might be useful to push the current state of `padding` to our own CSS classes and DOM elements which we use for nesting `NavigationControl` and `AttributionControl`
![Screen Shot 2019-08-20 at 3 21 37 PM](https://user-images.githubusercontent.com/1449257/63388722-3aa8e480-c35e-11e9-8bc0-943dfb3a4e7f.png)
`top`,`left`, `right` and `bottom` attributes are always 0, i think we can update them from within `Map`.

Having `Map` be the only API entrypoint seems more intuitive to me, with plugins automatically respecting it as well.

Reply:
Here's a demo of what I mean.

![2019-08-20 19 33 01](https://user-images.githubusercontent.com/1449257/63398128-71442680-c381-11e9-80f3-d4c98d48cc48.gif)